src/mixer/audio.ts

```diff
@@ -55,16 +55,73 @@ class Player extends ((PlayerEmitter as unknown) as { new (): EventEmitter }) {
     this.wavesurfer.drawBuffer();
   }
 
+  setCurrentTime(secs: number) {
+    this.wavesurfer.setCurrentTime(secs);
+  }
+
   setIntro(duration: number) {
+    if ("intro" in this.wavesurfer.regions.list) {
+      this.wavesurfer.regions.list.intro.end = duration;
+      this.redraw();
+      return;
+    }
+
     this.wavesurfer.addRegion({
       id: "intro",
       resize: false,
+      drag: false,
       start: 0,
       end: duration,
-      color: "rgba(125,0,255, 0.12)",
+      color: "rgba(125,0,255, 0.3)",
     });
   }
 
+  setCue(startTime: number) {
+    const duration = this.wavesurfer.getDuration();
+    const cueWidth = 0.01 * duration; // Cue region marker to be 1% of track length
+    if ("cue" in this.wavesurfer.regions.list) {
+      this.wavesurfer.regions.list.cue.start = startTime;
+      this.wavesurfer.regions.list.cue.end = startTime + cueWidth;
+      this.redraw();
+      return;
+    }
+
+    this.wavesurfer.addRegion({
+      id: "cue",
+      resize: false,
+      drag: false,
+      start: startTime,
+      end: startTime + cueWidth,
+      color: "rgba(0,100,0, 0.8)",
+    });
+  }
+
+  setOutro(startTime: number) {
+    if ("outro" in this.wavesurfer.regions.list) {
+      // If the outro is set to 0, we assume that's no outro.
```

Comment:
Is there a way to remove an outro? Set it to zero, I presume?

Reply:
Hmm, not the most obvious, but yes. Will try and make a reset button somehow

Reply:
Right-click?

Reply:
Once I figure out how to do this properly with more time, i'll do this in another small PR.